src/AllocationBoundsInference.cpp

```diff
@@ -86,6 +86,17 @@ class AllocationInference : public IRMutator {
                 max = b[i].max;
                 extent = simplify((max - min) + 1);
             }
+            if (bound.modulus.defined()) {
+                internal_assert(bound.remainder.defined());
+                min -= bound.remainder;
+                min = (min / bound.modulus) * bound.modulus;
+                min += bound.remainder;
+                Expr max_plus_one = max + 1;
+                max_plus_one -= bound.remainder;
+                max_plus_one = ((max_plus_one + bound.modulus - 1) / bound.modulus) * bound.modulus;
+                max_plus_one += bound.remainder;
+                extent = simplify(max_plus_one - min);
```

Comment:
Alas, it generates the same amount of code, just very slightly reshuffled.

On Tue, Jul 19, 2016 at 3:37 PM, Dillon Sharlet <notifications@github.com>
wrote:

> In src/AllocationBoundsInference.cpp
> <https://github.com/halide/Halide/pull/1392#discussion_r71437126>:
>
> > @@ -86,6 +86,17 @@ class AllocationInference : public IRMutator {
> >                  max = b[i].max;
> >                  extent = simplify((max - min) + 1);
> >              }
> > +            if (bound.modulus.defined()) {
> > +                internal_assert(bound.remainder.defined());
> > +                min -= bound.remainder;
> > +                min = (min / bound.modulus) * bound.modulus;
> > +                min += bound.remainder;
> > +                Expr max_plus_one = max + 1;
> > +                max_plus_one -= bound.remainder;
> > +                max_plus_one = ((max_plus_one + bound.modulus - 1) / bound.modulus) * bound.modulus;
> > +                max_plus_one += bound.remainder;
> > +                extent = simplify(max_plus_one - min);
>
> Ah, of course, my suggestion was wrong, but I think the idea of computing
> the aligned extent directly should work with rounding up, and I think it
> would generate simpler code.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/halide/Halide/pull/1392/files/2cba09eb217797c8f39846ba246d0406c15ab993#r71437126>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfdRo8IQfH0ezcu5sfnDPHKuETeHFy1ks5qXVGSgaJpZM4JQMmv>
> .
>